src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -2015,7 +2023,10 @@ def copy(self, new_sorting_analyzer, unit_ids=None):
             new_extension.data = self.data
         else:
             new_extension.data = self._select_extension_data(unit_ids)
-        new_extension.run_info = self.run_info.copy()
+        if self.run_info is not None:
```

Comment:
If you `import copy from copy` you can use `copy(self.run_info)` which returns `None` if `self.run_info` is `None`, which would save some lines of code and indentations here, at line 2047 and line 2268.